students/JerryH/Lesson08/circle.py

```diff
@@ -0,0 +1,57 @@
+import math
+
+class Circle:
+
+    def __init__(self, radius):
+        self.radius = radius
+        self._area = radius ** 2 * math.pi
+
+    @property
+    def diameter(self):
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        """ Read Only """
+        return self._area
+
+    @property
+    def from_diameter(self):
+        return self.diameter
+
+    #@from_diameter.setter
+    def from_diameter(self, diameter):
```

Comment:
(step 5) your `from_diameter` should be an alternate constructor, this means that it should be a class method:
```
@classmethod
def from_diameter(cls, diameter):
    return cls(diameter/2)
```
above will return an instance of Circle class by passing in radius derived from diameter